### ISSUE TITLE
unary_union on emtpy geometry segfaults when shapely is built against GEOS 3.10.1

### ISSUE THREAD

**Issue by ungarj** [shapely/shapely#1289]
## Expected behavior and actual behavior.

`unary_union` on empty geometry should result in an empty geometry. I tested it with GEOS 3.7.1 and it worked properly back then. See the docker example below.

## Steps to reproduce the problem.

```
FROM python:3.8-slim-buster as builder

RUN apt-get update \
    && apt install -y --no-install-recommends \
    cmake \
    build-essential \
    wget \
    && rm -rf /var/lib/apt/lists/*

WORKDIR /tmp

ENV GEOS_VERSION=3.10.1
# ENV GEOS_VERSION=3.7.1

# download source code
RUN wget -q https://download.osgeo.org/geos/geos-${GEOS_VERSION}.tar.bz2

RUN tar -xjf geos-${GEOS_VERSION}.tar.bz2 \
    && cd geos-${GEOS_VERSION} \
    && ./configure --prefix=/usr/local \
    && echo "building geos ${GEOS_VERSION}..." \
    && make --quiet -j${CPUS} && make --quiet install

RUN ldconfig

RUN pip install --upgrade pip \
    && pip install --no-binary :all: shapely

# if successful, this prints OK
RUN python -c "from shapely.geometry import box; from shapely.ops import unary_union; unary_union([box(0, 0, 0, 0).buffer(0)]); print('OK')"
```

## Operating system

linux

## Shapely version and provenance

1.8.0 installed from PyPI using pip and `--no-binary :all:` with GEOS 3.10.1 installed

**Comment by jorisvandenbossche** [comment]
@ungarj Thanks for the report!

I can reproduce this with the `shapely` 1.8 package from conda-forge, which is built with GEOS 3.10:

```python
In [1]: from shapely.geometry import box

In [2]: poly = box(0, 0, 0, 0).buffer(0)

In [3]: print(poly)
POLYGON EMPTY

In [4]: from shapely.ops import unary_union

In [5]: unary_union(poly)
Segmentation fault (core dumped)
```

**Comment by jorisvandenbossche** [comment]
Remembering now, we actually already had a bug report about this before (https://github.com/shapely/shapely/issues/1230), which triggered this to be fixed in GEOS (https://github.com/libgeos/geos/pull/502), which will be included in the upcoming GEOS 3.10.2

Going to close this as a duplicate of #1230

**Comment by ungarj** [comment]
Cool, thanks for the quick response, @jorisvandenbossche!
